spec/unit/resource_spec.rb

```diff
@@ -803,6 +803,72 @@ def self.name
     end
   end
 
+  describe "when using resource partials" do
+    let(:resource_using_core_partial) do
+      Class.new(Chef::Resource) do
+        use "core::core_partial"
+      end
+    end
+
+    let(:resource_using_cookbook_partial) do
+      Class.new(Chef::Resource) do
+        use "my_local_partial"
+      end
+    end
+
+    let(:resource_using_nested_partials) do
+      FakeCaller = Struct.new(:label, :path, keyword_init: true)
+      NonDynamicResource = Chef::Resource.dup
+
+      # Fake a caller_locations array, as RSpec uses a different execution path
+      NonDynamicResource.define_singleton_method(:caller_locations) do
+        [
+          FakeCaller.new(label: "use", path: File.join(__dir__, "resource.rb")),
+          FakeCaller.new(label: "noise", path: File.join(__dir__, "no_file")),
+          FakeCaller.new(label: "class_from_file", path: File.join(__dir__, "_level2_partial.rb")),
+          FakeCaller.new(label: "noise", path: File.join(__dir__, "no_file")),
+          FakeCaller.new(label: "class_from_file", path: File.join(__dir__, "_level1_partial.rb")),
+        ]
+      end
+
+      Class.new(NonDynamicResource) do
+        use "level1_partial"
+      end
+    end
+
+    it "correcly includes a core partial" do
+      expected_path = File.expand_path(File.join(__dir__, "../..", "lib/chef", "resource/_core_partial.rb"))
+      partial = "property :addon_property, default: true"
+
+      expect(IO).to receive(:read).with(expected_path).and_return(partial)
+      expect(resource_using_core_partial.properties.keys).to include(:addon_property)
+    end
+
+    it "correctly includes a cookbook partial" do
+      expected_path = File.expand_path(File.join(__dir__, "_my_local_partial.rb"))
+      partial = "property :addon_property, default: true"
+
+      expect(IO).to receive(:read).with(expected_path).and_return(partial)
+      expect(resource_using_cookbook_partial.properties.keys).to include(:addon_property)
+    end
+
+    it "correctly includes nested partials" do
```

Comment:
I still think we should address my comment here:
https://github.com/chef/chef/pull/13092#issuecomment-1238564602

Basically I think we need a test that covers more-than-2 to make sure we're not just incorrectly supporting base and base+1, but really supporting recursive includes. Your old cold passed this test, but wasn't quite right, and your new code, *I think* works, but the test to validate it fully and make sure no one comes along and "over optimizes" it later is important.